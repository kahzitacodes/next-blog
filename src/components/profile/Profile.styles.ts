import Image from 'next/image'
import tw from 'tailwind-styled-components'

export const Wrapper = tw.div`
  flex
  gap-6
  flex-col
  items-center
  md:flex-row
`

export const ProfileImage = tw(Image)`
  w-20
  self-start
  rounded-full
  border-2
  border-white
  md:self-center
`

export const Content = tw.div`
  flex
  flex-col
  gap-1
  justify-center
`

export const Title = tw.h1`
  text-3xl
  font-bold
`

export const Subtitle = tw.p`
  text-sm
  text-gray-300
`
