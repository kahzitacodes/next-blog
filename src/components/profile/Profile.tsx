import Image from 'next/image'

import { PropTypes } from './Profile.types'
import * as S from './Profile.styles'

export const Profile = (props: PropTypes) => {
  const { avatar, name, description } = props

  return (
    <S.Wrapper {...props}>
      <Image
        src={avatar}
        alt="profile image"
        title="profile image"
        width={100}
        height={100}
        className="w-20 self-start rounded-full border-2 border-white md:self-center"
        priority
      />
      <S.Content>
        <S.Title>{name}</S.Title>
        <S.Subtitle>{description}</S.Subtitle>
      </S.Content>
    </S.Wrapper>
  )
}
