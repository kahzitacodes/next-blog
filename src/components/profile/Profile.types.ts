export type PropTypes = {
  avatar: string
  name: string
  description?: string
}
