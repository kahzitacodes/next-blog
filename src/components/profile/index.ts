export * from './Profile'
export type { PropTypes as ProfileProps } from './Profile.types'
