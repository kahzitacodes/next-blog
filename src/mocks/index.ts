export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
