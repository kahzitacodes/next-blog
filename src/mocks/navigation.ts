import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
