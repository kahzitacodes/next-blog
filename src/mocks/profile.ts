import type { ProfileProps } from '@/components'

export const profileMock: ProfileProps = {
  avatar: '/assets/images/avatar.jpeg',
  name: 'Karine Pereira',
  description: 'Lorem ipsum dolor sit amet.'
}
