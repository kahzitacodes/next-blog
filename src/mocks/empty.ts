import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
