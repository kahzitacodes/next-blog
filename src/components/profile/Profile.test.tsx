import { render, screen } from '@testing-library/react'
import { Profile } from '.'
import { profileMock } from '@/mocks/profile'

describe('<Profile />', () => {
  it('should render successfully', () => {
    const { container } = render(<Profile {...profileMock} />)

    expect(
      screen.getByRole('heading', { name: /Karine Pereira/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/Lorem ipsum dolor sit amet./i)).toBeInTheDocument()

    expect(screen.getByAltText(/profile image/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
