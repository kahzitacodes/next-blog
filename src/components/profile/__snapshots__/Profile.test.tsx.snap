// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render successfully 1`] = `
<div
  avatar="/assets/images/avatar.jpeg"
  class="flex gap-6 flex-col items-center md:flex-row"
  description="Lorem ipsum dolor sit amet."
  name="Karine Pereira"
>
  <img
    alt="profile image"
    class="w-20 self-start rounded-full border-2 border-white md:self-center"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    height="100"
    src="/_next/image?url=%2Fassets%2Fimages%2Favatar.jpeg&w=256&q=75"
    srcset="/_next/image?url=%2Fassets%2Fimages%2Favatar.jpeg&w=128&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Favatar.jpeg&w=256&q=75 2x"
    style="color: transparent;"
    title="profile image"
    width="100"
  />
  <div
    class="flex flex-col gap-1 justify-center"
  >
    <h1
      class="text-3xl font-bold"
    >
      Karine Pereira
    </h1>
    <p
      class="text-sm text-gray-300"
    >
      Lorem ipsum dolor sit amet.
    </p>
  </div>
</div>
`;
